helm-charts/basehub/values.yaml

```
@@ -757,3 +759,71 @@ jupyterhub:
             # Customize list of profiles dynamically, rather than override options form.
             # This is more secure, as users can't override the options available to them via the hub API
             c.KubeSpawner.profile_list = custom_profile_list
+
+      06-salted-username: |
+        # Allow anonymizing username to not store *any* PII
+        import json
+        import os
+        import base64
+        import hashlib
+        from z2jh import get_config
+
+        USERNAME_DERIVATION_PEPPER = bytes.fromhex(os.environ['USERNAME_DERIVATION_PEPPER'])
+
+        def salt_username(authenticator, handler, authentication):
```

Just a small personal preference to use the same name for the dict info returned by the `authenticate` call [as it's in `oauthenticator`](https://github.com/jupyterhub/oauthenticator/blob/8ff022c0479893aede07714e522f5f48f97141d9/oauthenticator/oauth2.py#L973-L978) if we choose to not document the parameters.

```suggestion
        def salt_username(authenticator, handler, auth_model):
```

---

I now realized jupyterhub calls it `the authentication dict` while oauthenticator `auth_model`. Please feel free to ignore my suggestion.

---

`auth_model` is a better name, we should change upstream to that too! I'll change it here.